### PR TITLE
refactor(Form): input events

### LINKED ISSUE #1: Handle native form events like `@change`

### LINKED ISSUE #100: [Form] Input events & Validation
The goal is to clarify when form validation is triggered for each inputs and determine which events are emitted by input components.

Here's a comparison between events for `native`, `radix-vue` and `@nuxt/ui@3` elements:

### `Input` & `Textarea`
|          | `native` | `@nuxt/ui` |
|----------|----------|------------|
| `input`  | x        | x          |
| `change` | x        | x          |
| `blur`   | x        | x          |

### `Checkbox` & `Switch`
|          | `native` | `radix-vue` | `@nuxt/ui` |
|----------|----------|-------------|------------|
| `input`  | x        | x           | x          |
| `change` | x        |             |            |
| `blur`   |          |             |            |

### `RadioGroup`
|          | `native` | `radix-vue` | `@nuxt/ui` |
|----------|----------|-------------|------------|
| `input`  | x        | x           | x          |
| `change` | x        |             |            |
| `blur`   |          |             |            |

### `Slider`
|          | `native` | `radix-vue`      | `@nuxt/ui` |
|----------|----------|------------------|------------|
| `input`  | x        | x                | x          |
| `change` | x        | x `@valueCommit` | x          |
| `blur`   |          |                  |            |

### `Select`
|          | `native` | `radix-vue` | `@nuxt/ui` |
|----------|----------|-------------|------------|
| `input`  | x        | x           | x          |
| `change` | x        |             |            |
| `blur`   |          |   x         |   x        |

### `SelectMenu`
|          | `radix-vue` | `@nuxt/ui` |
|----------|-------------|------------|
| `input`  | x           | x          |
| `change` |             |           |
| `blur`   | x           | x          |

### `InputMenu`
|          | `radix-vue` | `@nuxt/ui` |
|----------|-------------|------------|
| `input`  | x           | x          |
| `change` |             |           |
| `blur`   | x           | x          |

There's a difference in how the `change` event is handled in regular HTML elements versus certain components in `radix-vue` like `Checkbox`, `Select`, and `RadioGroup`. Since both `input` and `change` events act the same for those elements, we can just use the `radix-vue` way and ditch the `change` event. Or, we could add these events to `@nuxt/ui` to match how native components work.

## Form validation

Regarding validation with the `Form` component:
- Validation for `input` occurs as you type. For example, if you're filling in a text field, validation happens automatically as you type each character.
- Validation for `change` occurs when you commit to a value. For `Input` elements, this means when you click away from the field after typing something.
- Validation for `blur` happens when you click away from a field. So, if you type something in a field and then click anywhere else on the page, validation kicks in.

### Approach 1
We choose to emit an `input` event on top of the `change` for all components. This seems to be the expected behaviour, but requires to de-duplicate validation events since a lot of components will emit both `change` and `input` events at the same time.

|              | `input` | `change` | `blur` |
|--------------|---------|----------|--------|
| `Input`      | x       | x        | x      |
| `Textarea`   | x       | x        | x      |
| `Checkbox`   | x       | x        |        |
| `Toggle`     | x       | x        |        |
| `Select`     | x       | x        | x      |
| `SelectMenu` | x       | x        | x      |
| `InputMenu`  | x       | x        | x      |
| `Slider`     | x       | x        |        |
| `RadioGroup` | x       | x        |        |

### Approach 2
The `input` event is emitted only when there's a notable difference with the `change` event. This does not require to de-duplicate input events, the downside is that using `:validate-on="['input']"` won't trigger validation for a lot of components.

|              | `input` | `change` | `blur` |
|--------------|---------|----------|--------|
| `Input`      | x       | x        | x      |
| `Textarea`   | x       | x        | x      |
| `Checkbox`   |         | x        |        |
| `Switch`     |         | x        |        |
| `Select`     |         | x        |  x    |
| `SelectMenu` |         | x        | x      |
| `InputMenu`  | x       | x        | x      |
| `Slider`     | x       | x        |        |
| `RadioGroup` |         | x        |        |

### PR DESCRIPTION
**Pull request by romhml**
Resolves #1 and resolves #100

### CODE REVIEW

**Comment by benjamincanac** [comment]
@romhml Can I let you fix the conflicts?

**Comment by benjamincanac** [comment]
@romhml What is left to do on this?

**Comment by romhml** [comment]
I wrote tests first this weekend to ensure this matched the specs defined in #100. Now I need to fix each components to match these tests. I'll find some time to finish it this week.

File: src/runtime/components/Select.vue [hunk 1]
```diff
@@ -113,8 +117,22 @@ const ui = computed(() => tv({ extend: select, slots: props.ui })({
 
 const groups = computed(() => props.items?.length ? (Array.isArray(props.items[0]) ? props.items : [props.items]) as SelectItem[][] : [])
 
+function onUpdate(value: any) {
+  const event = new Event('change', { target: { value } })
```
Comment (romhml) [inline src/runtime/components/Select.vue:121]: This is a bit sketchy but allows these change events to be consistent among native and custom components

File: src/runtime/components/Select.vue [hunk 2]
```diff
@@ -113,8 +117,22 @@ const ui = computed(() => tv({ extend: select, slots: props.ui })({
...
+  emits('change', event)
+
+  emitFormChange()
+  emitFormInput()
```
Comment (romhml) [inline src/runtime/components/Select.vue:125]: Will validate the form twice. I'll look into implementing a `dirty` check on the form component to ignore validation requests when the value has not changed to improve performances.

File: src/runtime/components/RadioGroup.vue [hunk 3]
```diff
@@ -91,17 +93,27 @@ const normalizedItems = computed(() => {
   if (!props.items) return []
   return props.items.map(normalizeItem)
 })
+
+const rootRef = ref()
```
Comment (benjamincanac) [inline src/runtime/components/RadioGroup.vue]: What is `rootRef` used for here?

Comment (romhml) [reply src/runtime/components/RadioGroup.vue]: This is a leftover from me trying to pass the target element in the change event. I'll clean this up.

**Review by benjamincanac — APPROVED** [review]

**Pull request merged**